src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -1553,16 +1565,21 @@ internal void OnSequencePointHit(FunctionContext functionContext)
             {
                 if (functionContext._breakPoints[functionContext._currentSequencePointIndex])
                 {
-                    var breakpoints = (from breakpoint in functionContext._boundBreakpoints
-                                       where
-                                           breakpoint.SequencePointIndex == functionContext._currentSequencePointIndex &&
-                                           breakpoint.Enabled
-                                       select breakpoint).ToList<Breakpoint>();
-
-                    breakpoints = TriggerBreakpoints(breakpoints);
-                    if (breakpoints.Count > 0)
+                    if (functionContext._boundBreakpoints.TryGetValue(functionContext._currentSequencePointIndex, out var bps))
                     {
-                        StopOnSequencePoint(functionContext, breakpoints);
+                        var breakpoints = (from breakpoint in bps
+                                        where
+                                            breakpoint.Enabled
+                                        select breakpoint).ToList<Breakpoint>();
+
+                        if (breakpoints.Count > 0)
+                        {
+                            breakpoints = TriggerBreakpoints(breakpoints);
+                            if (breakpoints.Count > 0)
+                            {
+                                StopOnSequencePoint(functionContext, breakpoints);
+                            }
+                        }
```

Comment:
I know it's just a style thing and it's not your fault @nohwnd, but I don't love this double count check.

Ideally we could just put the check inside `TriggerBreakpoints`.

If I were really trying to make this suit my desired style, I'd make them into extension methods:

```csharp
breakpoints.Trigger().StopOnSequencePoint(functionContext);
```

Reply:
I don't love it either, but it prevents creating another list in TriggerBreakpoints when we call it with empty list of breakpoints. And moving the check into TriggerBreakpoints makes this code path less obvious. 

I also don't love that the breakpoints variable is reused, but I went for the minimal amount of changes in this PR. If you insist on changing it I can do it. As you say it's style related. Should I make the change?

Reply:
I think it's fine as is for now